backend/src/main/java/ai/giskard/service/ProjectService.java

```diff
@@ -93,7 +127,134 @@ public void delete(Long id) {
+
+        // Storage of each element in the db
+         models.forEach(model -> {
+             Long formerId = model.getId();
+             Path formerModelPath = temporaryExportDir.resolve("models").resolve(FileLocationService.createZSTname("model_", formerId));
+             Path formerRequirementsModelPath = temporaryExportDir.resolve("models").resolve(FileLocationService.createTXTname("model-requirements_", formerId));
+             try {
+                 model.setProject(savedProject);
+                 FileInputStream modelStream = new FileInputStream(formerModelPath.toFile());
+                 FileInputStream modelRequirementStream = new FileInputStream(formerRequirementsModelPath.toFile());
+                 ProjectModel savedModel = fileUploadService.uploadModel(model, modelStream, modelRequirementStream);
+                 mapFormerNewIdModelDataset.put(formerId, savedModel.getId());
+             } catch (IOException e) {
+                 throw new GiskardRuntimeException("Couldn't upload the models");
+             }
+         });
+
+         datasets.forEach(dataset -> {
+             Long formerId = dataset.getId();
+             Path formerDatasetPath = temporaryExportDir.resolve("datasets").resolve(FileLocationService.createZSTname("data_", formerId));
+             try{
+                 dataset.setProject(savedProject);
+                 FileInputStream datasetStream = new FileInputStream(formerDatasetPath.toFile());
+                 Dataset savedDataset = fileUploadService.uploadDataset(savedProject, dataset.getName(), dataset.getFeatureTypes(), dataset.getColumnTypes(), dataset.getTarget(), datasetStream);
+                 mapFormerNewIdModelDataset.put(formerId, savedDataset.getId());
+             }
+             catch(IOException e){
+                 throw new GiskardRuntimeException("Couldn't upload the datasets");
+             }
+         });
+
+         feedbacks.forEach(feedback -> {
+             feedback.setProject(savedProject);
+             feedback.setDataset(datasetRepository.getById(mapFormerNewIdModelDataset.get(feedback.getDataset().getId())));
+             feedback.setModel(modelRepository.getById(mapFormerNewIdModelDataset.get(feedback.getModel().getId())));
+             Feedback savedFeedback = feedbackRepository.save(feedback);
+             feedback.getFeedbackReplies().forEach(reply -> {
+                 reply.setFeedback(savedFeedback);
+                 feedbackReplyRepository.save(reply);
+             });
+             feedbackRepository.save(savedFeedback);
+         });
+
+         testSuites.forEach(testSuite -> {
+             testSuite.setProject(savedProject);
+             testSuite.setActualDataset(datasetRepository.getById(mapFormerNewIdModelDataset.get(testSuite.getActualDataset().getId())));
+             testSuite.setReferenceDataset(datasetRepository.getById(mapFormerNewIdModelDataset.get(testSuite.getReferenceDataset().getId())));
```

Comment:
`getReferenceDataset()` and `getActualDataset()` can return null resulting into NullPointerException